core/escrow/escrow.go

```diff
@@ -373,40 +373,54 @@ func SyncContractPayoutStatus(ctx context.Context, n *core.IpfsNode,
 	err = grpc.EscrowClient(cfg.Services.EscrowDomain).WithContext(ctx,
 		func(ctx context.Context,
 			client escrowpb.EscrowServiceClient) error {
+			in := &escrowpb.SignedContractIDBatch{
+				Data: &escrowpb.ContractIDBatch{
+					Address: pkBytes,
+				},
+			}
 			for _, c := range cs {
-				in := &escrowpb.SignedContractID{
-					Data: &escrowpb.ContractID{
+				in.Data.ContractId = append(in.Data.ContractId, c.GuardContract.ContractId)
+			}
+			sign, err := crypto.Sign(n.PrivateKey, in.Data)
+			if err != nil {
+				log.Error("sign contractID error:", err)
+				return err
+			}
+			in.Signature = sign
+			sb, err := client.GetPayOutStatusBatch(ctx, in)
+			if err != nil {
+				log.Error("get payout status batch error:", err)
+				return err
+			}
+			if len(sb.Status) != len(cs) {
+				return fmt.Errorf("payout status batch returned wrong length of contracts, need %d, got %d",
+					len(cs), len(sb.Status))
+			}
+			for i, c := range cs {
+				s := sb.Status[i]
+				// Set dummy payout status on invalid id or error msg
+				if s.ContractId != c.GuardContract.ContractId {
+					s = &escrowpb.PayoutStatus{
 						ContractId: c.GuardContract.ContractId,
-						Address:    pkBytes,
-					},
-				}
-				sign, err := crypto.Sign(n.PrivateKey, in.Data)
-				if err != nil {
-					log.Error("sign contractID error:", err)
-					continue
-				}
-				in.Signature = sign
-				s, err := client.GetPayOutStatus(ctx, in)
-				if err != nil {
-					// It's possible contract is in initial state and does not
-					// have actual payout status yet
-					if err.Error() != payoutNotFoundErr {
-						log.Errorf("state: [%s], get payout status error: %v", c.GuardContract.State, err)
+						ErrorMsg:   "mistmatched contract id",
```

Comment:
On current implementation, this error should not occur as escrow try to give result in same index.  But if this error really happens(escrow met some error), maybe we can use double pointer to match them, performance is not good, but at least we got the result.